Notify GitHub of Jenkins build status

djbingham opened the pull request:
https://eaflood.atlassian.net/browse/PSD-78

Commit:
Report Jenkins build status to GitHub

https://eaflood.atlassian.net/browse/PSD-78

Commit:
Compute commit SHA and PR num only once per build

Commit:
Set variables before attempting to notify GitHub

Commit:
Correct namespace delete command in Jenkinsfile

markysoft reviewed (dismissed):
LGTM, should fix the error on the master deployment build cleaning up the namespace too.

markysoft commented on Jenkinsfile:
```diff
@@ -94,37 +117,58 @@ def publishChart(imageName) {
+    }
+    if (pr != '') {
+      stage('Helm install') {
+        def extraCommands = "--values ./helm/$repoName/jenkins-aws.yaml"
```
The deployment needs to override the service URLs as there are no FFC services available in the `default` namespace. The namespace for master deployments is `ffc-demo` in the AWS cluster.

djbingham replied:
Good spot. In fact, if the services are deployed to the same namespace then we don't need to specify the namespace at all in URLs, which seems a little more flexible. I've tested this on my local K8s cluster using `wget -qO- http://ffc-demo-user-service/healthz` from within an API Gateway container in the same namespace as the user service.

I've made that change in `values.yaml` so that services can communicate regardless of what namespace they're deployed to (as long as they share a namespace, which was the case when specifying `default` anyway).

markysoft replied:
Specifing the `ffc-demo` namespace is preferred as there should always be services available in that namespace for testing a PR, once all the service are deployed. 
Removing the namespace altogether would mean every PR would need services in the PR namespace, which will not be available.

djbingham replied:
Ah, that's a good idea. Updated.

markysoft reviewed (changes requested):
API Urls are pointing to the default namespace.

Commit:
Declare variables used in Jenkinsfile

New variables: repoUrl and commitSha

Matthew-Collins-Defra reviewed (dismissed):
Your Jenkinsfile is missing the
`def repoUrl = ''`
`def commitSha = ''`
from the top of the file, so will not run.

djbingham commented:
> Your Jenkinsfile is missing the
> `def repoUrl = ''`
> `def commitSha = ''`
> from the top of the file, so will not run.

It actually had run and passed but I have added the declarations anyway. Interesting that Jenkins doesn't seem to mind variables not being declared.

Commit:
Remove "Build production image" step from Jenkins

This was redundant as the "Push container image" step actually builds
the image first anyway.

Commit:
Correct namespace in port-forwarding instruction

Commit:
Correct service URLs in Jenkins deployment

Commit:
Set service URLs the same for all deployments

If services are deployed in the same namespace, they don't need to
specify that namespace in URLs to communicate.

Commit:
Remove unnecessary comment from Jenkinsfile

Commit:
Set service URLs to include ffc-demo namespace

PR deployments can interact with a shared set of services in the
ffc-demo namespace, rather than requiring a full stack deployed in each
PR namespace for each service.
